src/indexset.jl

```diff
@@ -1,64 +1,43 @@
 
 # TODO: extend type restriction to `IndexT <: Union{<: Index, <: IndexVal}`
 struct IndexSet{IndexT <: Index}
-  data::Tuple{Vararg{IndexT}}
+  data::Vector{IndexT}
 
   function IndexSet{IndexT}(data) where {IndexT}
     @debug_check begin
       if !allunique(data)
         error("Trying to create IndexSet with collection of indices $data. Indices must be unique.")
       end
     end
-    return new{IndexT}(data)
+    return new{IndexT}([data...])
```

Comment:
I guess this is something to think about. This would copy every time an `IndexSet` gets constructed, i.e. `IndexSet([i, i', i''])` would copy. I'm not sure this is necessary, since the goal is for `IndexSet` to have immutable semantics anyway. I always go back and forth about whether or not constructors should make views or copies, but I think we could think about `IndexSet` as a light wrapper around a `Vector`. Also it will be very common to build an `IndexSet` by first building a `Vector{Index}` so it would be unfortunate if it copied every time...

Reply:
Presumably you added this in order to handle more generic inputs besides `Vector`? In that case we could have an inner constructor that explicitly accepts `Vector` (without copying) and then an outer constructor that, if the input isn't a `Vector`, converts to `Vector` with `collect`.